code/mission/missionparse.cpp

```diff
@@ -4869,19 +4869,32 @@ void parse_event(mission * /*pm*/)
 
 	if ( optional_string("+Repeat Count:")){
 		stuff_int( &(event->repeat_count) );
+
+		// sanity check on the repeat count variable
+		// _argv[-1] - negative repeat count is now legal; means repeat indefinitely.
+		if ( event->repeat_count == 0 ){
+			Warning(LOCATION, "Repeat count for mission event %s is 0.\nMust be >= 1 or negative!  Setting to 1.", event->name );
+			event->repeat_count = 1;
+		}
 	} else {
 		event->repeat_count = 1;
 	}
 
 	if ( optional_string("+Trigger Count:")){
 		stuff_int( &(event->trigger_count) );
+		event->flags |= MEF_USING_TRIGGER_COUNT; 
+
 		// if we have a trigger count but no repeat count, we want the event to loop until it has triggered enough times
 		if (event->repeat_count == 1) {
 			event->repeat_count = -1;
 		}
-		event->flags |= MEF_USING_TRIGGER_COUNT; 
-	} 
-	else {
+
+		// sanity check on the trigger count variable
+		if ( event->trigger_count <= 0 ){
+			Warning(LOCATION, "Trigger count for mission event %s is <= 0.\nMust be >= 1!  Setting to 1.", event->name );
+			event->trigger_count = 1;
+		}
```

Comment:
This is wrong; a trigger count of -1 means "infinite", just like a repeat count of -1.

Reply:
Ok.